BE/src/main/java/com/example/techstore/service/impl/SlideServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+import com.example.techstore.exception.NotFoundException;
+import com.example.techstore.repository.ProductRepository;
+import com.example.techstore.repository.SlideRepository;
+import com.example.techstore.service.SlideService;
+import com.example.techstore.util.PaginationUtil;
+import com.example.techstore.util.UploadFileUtil;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class SlideServiceImpl implements SlideService {
+    private final SlideRepository slideRepository;
+    private final SlideMapper slideMapper;
+    private final ProductRepository productRepository;
+    private final UploadFileUtil uploadFileUtil;
+
+    @Override
+    public SlideDto getById(String id) {
+        Slide slide = slideRepository.findById(id)
+                .orElseThrow(() -> new NotFoundException(ErrorMessage.Slide.ERR_NOT_FOUND_ID, new String[]{id}));
+        return slideMapper.mapSlideToSlideDto(slide);
+    }
+
+    @Override
+    public PaginationResponseDto<SlideDto> getAll(PaginationFullRequestDto paginationFullRequestDto) {
+        Pageable pageable = PaginationUtil.buildPageable(paginationFullRequestDto, SortByDataConstant.SLIDE);
+        Page<Slide> slidePage = slideRepository.getAll(pageable);
+        PagingMeta meta = PaginationUtil
+                .buildPagingMeta(paginationFullRequestDto, SortByDataConstant.SLIDE, slidePage);
+
+        List<SlideDto> slideDtoList = slideMapper.mapSlideToSlideDto(slidePage.getContent());
+        return new PaginationResponseDto<>(meta, slideDtoList);
+    }
+
+    @Override
+    public PaginationResponseDto<SlideDto> getByStatus(PaginationFullRequestDto paginationFullRequestDto, Boolean status) {
+        Pageable pageable = PaginationUtil.buildPageable(paginationFullRequestDto, SortByDataConstant.SLIDE);
+        Page<Slide> slidePage = slideRepository.getByStatus(status, pageable);
+
+        PagingMeta meta = PaginationUtil.buildPagingMeta(paginationFullRequestDto, SortByDataConstant.SLIDE, slidePage);
+        List<SlideDto> slideDtoList = slideMapper.mapSlideToSlideDto(slidePage.getContent());
+        return new PaginationResponseDto<>(meta, slideDtoList);
+    }
+
+    @Override
+    public SlideDto create(SlideRequestDto createDto) {
+        Product product = productRepository.findById(createDto.getProductId())
+                .orElseThrow(() -> new NotFoundException(ErrorMessage.Product.ERR_NOT_FOUND_ID,
+                        new String[]{createDto.getProductId()}));
+
+        Slide check_position = slideRepository.findByPosition(createDto.getPosition());
+        if (check_position != null){
+            throw new NotFoundException(ErrorMessage.Slide.ERR_POSITION_ALREADY_EXIST);
```

Comment:
```suggestion
            throw new NotFoundException(ErrorMessage.Slide.ERR_POSITION_ALREADY_EXIST);
``` throw AlreadyException có sẵn